src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -3535,7 +3535,12 @@ internal Collection<PSObject> NewItem(string[] paths, string name, string type,
                             throw PSTraceSource.NewInvalidOperationException(SessionStateStrings.PathNotFound, targetPath);
                         }
 
-                        content = globbedTarget[0];
+                        // If the original target was a relative path, we want to leave it as relative if it did not require
+                        // globbing to resolve.
+                        if (WildcardPattern.ContainsWildcardCharacters(content.ToString()))
+                        {
+                            content = globbedTarget[0];
```

Comment:
We do globing then check that the globing is not needed - looks not efficient.

Reply:
I saw that, but left it after the globbing because the globbing is also needed to check the target provider on line 3523

Reply:
I see.
I feel that there is something wrong. :-(
Perhaps it should be internally in globbing?

Reply:
@SteveL-MSFT Are you sure that this fix should not be internally in globbing?

Reply:
Your suggestion is that the globber should return back the relative path if it doesn't match anything extra?  Seems like a big change that may have unintended side effects to other uses of the globber which currently returns absolute paths.

Reply:
Something similar is already working for completions. Although perhaps there was also a partial fix.

Reply:
It is not performance critical so I am ok.